### PR TITLE
Update to allow running outside a cloud

### PR DESCRIPTION
**Pull request by manics**
Previously this role was used in the IDR inside a segregated network where all servers are standardised. We now want to run this on a shared VLAN where the servers aren't standard, and where we'd ideally not expose the default docker ports. This requires:
- Additional custom configuration of targets to be scraped
- Ability to proxy a container directly using the internal Docker IP instead of using Docker port-forwarding
- Support for externally managed Docker host volumes

Tag: `0.5.0` new feature

Requires
- [x] https://github.com/manics/action-ansible-galaxy-publish to be transferred to `ome`
- [ ] `GALAXY_API_KEY` secret

### CODE REVIEW

File: README.md
Comment (joshmoore) [inline README.md:50]: Nice. Good to think about all roles as having potential outputs.

File: README.md
Comment (joshmoore) [inline README.md]: It probably would be nice to have some trivial example here.

File: tasks/main.yml
Comment (joshmoore) [inline tasks/main.yml:150]: Is there any potential for someone to mistakenly run two of these with the same volume? (If so, I assume that problem existed _before_ this PR as well)

File: templates/etc-prometheus-prometheus-yml.j2
Comment (joshmoore) [inline templates/etc-prometheus-prometheus-yml.j2:135]: to_json in a yaml file?

**Review by joshmoore — COMMENTED** [review]
Few minor points but generally MSTM.

File: tasks/main.yml
Comment (manics) [reply tasks/main.yml:150]: In theory they could, but that's not something we can easily fix at the role level without a lot more complexity. It's also a problem for any of our roles that assume write access by a process.

This is one example where K8s is better, since in a Helm chart you can easily namespace everything.

File: templates/etc-prometheus-prometheus-yml.j2
Comment (manics) [reply templates/etc-prometheus-prometheus-yml.j2:135]: `item` needs to be serialised in a YAML compatible format. Since it's part of an indented list the multiline `to_yaml` function doesn't work (the indent parameter only seems to set the nested indentation, not the top-level). `to_json` is a way to obtain one-line serialised YAML, since in general JSON is valid YAML (I think the only exception is white space after a `:` is required in YAML but not JSON)

Comment (joshmoore) [reply templates/etc-prometheus-prometheus-yml.j2:135]: Understood. I assume it's clearly working, just seemed surprising.

File: README.md
Comment (manics) [reply README.md]: I've added a link to the relevant section of the Prometheus docs.

**Comment by joshmoore** [comment]
Most recent changes seem fine. :+1: @jburel was working on a token key I think. So I'd defer to him on that.

**Comment by joshmoore** [comment]
(Hit the wrong button. Re-opening)

**Review by joshmoore — APPROVED** [review]
My :+1: from https://github.com/ome/ansible-role-prometheus/pull/9#issuecomment-763486712 still holds.

**Pull request merged**
